webxrlayers-1.bs

```diff
@@ -954,9 +940,10 @@ When the <dfn method for="XRSession" dfn-for="layers">updateRenderState(|newStat
 with "[=feature descriptor/layers=]" enabled, the user agent MUST run the following steps:
```

Comment:
This needs to be updated to patch the `update the pending layers state` algorithm.

Something like, "When the `update the pending layers state` algorithm needs to be called, instead call the `update the pending layers state with layers support` algorithm, defined below.

The algorithm should do a feature check, and also do all the other stuff below

Reply:
> This needs to be updated to patch the `update the pending layers state` algorithm.

That is what this code is doing.

> Something like, "When the `update the pending layers state` algorithm needs to be called, instead call the `update the pending layers state with layers support` algorithm, defined below.

Are you saying that it needs to be more explicit that it's replacing the other algorithm?

> The algorithm should do a feature check, and also do all the other stuff below

Isn't the feature check covered by "an XRSession session with "layers" enabled"?
What do you mean by "other stuff below"? All that needs to happen is to have an empty array.

Reply:
Right, the reason we introduced the hook is so that we don't have to replace the entire updateRenderState algorithm, just this part.

It's fine to also make the patch dynamic and dependent on the feature, but it's better if that's in the spec text IMO so that we can clearly define how the behavior is with and without the feature.

Reply:
OK. I just updated the patch and I think it's better. Please take a look and thanks for all your help!